internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/endToEnd/WriteComputeWithActiveActiveReplicationTest.java

```diff
@@ -0,0 +1,355 @@
+
+    sendStreamingRecord(dcSystemProducer.get(childDatacenters.get(1)), storeName, key1, ubKv1F1.build());
+    validatePersonV1Record(storeName, dc0RouterUrl, key1, "Bar", 42);
+    validatePersonV1Record(storeName, dc1RouterUrl, key1, "Bar", 42);
+
+    sendStreamingRecord(dcSystemProducer.get(childDatacenters.get(0)), storeName, key1, ubKv1All.build());
+    validatePersonV1Record(storeName, dc0RouterUrl, key1, "FooBar", 24);
+    validatePersonV1Record(storeName, dc1RouterUrl, key1, "FooBar", 24);
+
+    /*
+     * Case: Updates with lower timestamps should be ignored && updates with higher timestamps should be applied
+     * Steps:
+     * 1. Put full record containing two fields and timestamp t2. Outcome: full update applied
+     * 2. Update the first field with timestamp t1; here t1 < t2 and hence the update should be ignored
+     * 3. Update the second field with timestamp t3; here t3 > t2 and hence the update should be applied
+     * 4. Update the first field with timestamp t2; since t2 == t2 update should be applied?
+     */
+    String key2 = "key2";
+    GenericRecord val2 = new GenericData.Record(valueSchemaV1);
+    val2.put(PERSON_F1_NAME, "val2f1_b");
+    val2.put(PERSON_F2_NAME, 20);
+    VeniceObjectWithTimestamp val2T2 = new VeniceObjectWithTimestamp(val2, 2);
+
+    UpdateBuilder ub2 = new UpdateBuilderImpl(wcSchemaV1);
+    ub2.setNewFieldValue(PERSON_F1_NAME, "val2f1_a");
+    VeniceObjectWithTimestamp val2T1 = new VeniceObjectWithTimestamp(ub2.build(), 1);
+
+    UpdateBuilder ub3 = new UpdateBuilderImpl(wcSchemaV1);
+    ub3.setNewFieldValue(PERSON_F2_NAME, 40);
+    VeniceObjectWithTimestamp val2T3 = new VeniceObjectWithTimestamp(ub3.build(), 3);
+
+    UpdateBuilder ub2Prime = new UpdateBuilderImpl(wcSchemaV1);
+    ub2Prime.setNewFieldValue(PERSON_F1_NAME, "val2f1_b_prime");
+    VeniceObjectWithTimestamp val2T2Prime = new VeniceObjectWithTimestamp(ub2Prime.build(), 2);
+
+    sendStreamingRecord(dcSystemProducer.get(childDatacenters.get(0)), storeName, key2, val2T2);
+    validatePersonV1Record(storeName, dc0RouterUrl, key2, "val2f1_b", 20);
+    validatePersonV1Record(storeName, dc1RouterUrl, key2, "val2f1_b", 20);
+
+    // val2T1 update should be ignored since its timestamp t1 is lower than the existing timestamp t2 for name field
+    sendStreamingRecord(dcSystemProducer.get(childDatacenters.get(1)), storeName, key2, val2T1);
+
+    // val2T3 update should be applied since its timestamp t3 is greater than the existing timestamp t2 for age field
+    sendStreamingRecord(dcSystemProducer.get(childDatacenters.get(1)), storeName, key2, val2T3);
+    validatePersonV1Record(storeName, dc0RouterUrl, key2, "val2f1_b", 40);
+    validatePersonV1Record(storeName, dc1RouterUrl, key2, "val2f1_b", 40);
+
+    // val2T2Prime update should be applied since its timestamp t2 is equal to the existing timestamp t2 for name field
+    // TODO: is this expected behavior though? It seems this can cause divergence?
```

Comment:
Why?

Reply:
If we apply an update with the same timestamp as that of the current timestamp of the field then in the case of concurrent writers with the same timestamp the outcome will be non-deterministic. 

eg.
```
dc1:
writer1:  (x:2, ts:T)
writer2: (x:20, ts:T)
finalValueOfX=20

dc2:
writer2: (x:20, ts:T)
writer1:  (x:2, ts:T)
finalValueOfX=2
```

Do we use coloIds to break ties?

Reply:
I believe for non collection merging operation, it will basically compare the hashcode of the bytes of the whole value or a single field, bigger wins. For collection merging operation, I think you are right it uses colo ID as well. checkout SortBasedCollectionFieldOpHandler to see how it uses colo ID as well for breaking ties.

Reply:
I see. Thanks for the clarification.